tools/firmware.toit

```diff
@@ -833,18 +798,24 @@ class Envelope:
       throw "cannot open envelope - expected version $ENVELOPE_FORMAT_VERSION, was $version"
     return version
 
+class RelocatableImage:
+  id/uuid.Uuid
+  relocatable/ByteArray
+  constructor .id .relocatable:
+
 class ContainerEntry:
   name/string
-  content/ByteArray
+  image/RelocatableImage
   assets/ByteArray?
-  constructor .name .content --.assets:
+  constructor .name .image --.assets:
 
 class ImageHeader:
-  static MARKER_OFFSET_   ::= 0
-  static ID_OFFSET_       ::= 8
-  static METADATA_OFFSET_ ::= 24
-  static UUID_OFFSET_     ::= 32
-  static HEADER_SIZE_     ::= 48
+  static MARKER_OFFSET_        ::= 0
+  static ID_OFFSET_            ::= 8
+  static METADATA_OFFSET_      ::= 24
+  static UUID_OFFSET_          ::= 32
+  static SNAPSHOT_UUID_OFFSET_ ::= 48 + 7 * 2 * 4  // 7 tables and lists.
```

Comment:
Comments here and in program.h to link them together.

Reply:
Done.